### PR TITLE
refactor(mention-area): migrate mention-area to styled component

### PR DESCRIPTION
**Pull request by binkpitch**
#532 
This migrate mention area component to styled component.

Note that, I am not really sure about how to refractor this part.

` <Animated.View style={[{ ...this.props.style }, { height: this.state.height }]}>
       ...
   </Animated.View>`

### CODE REVIEW

File: src/components/mention-area.component.js [hunk 1]
```diff
@@ -1,36 +1,30 @@
...
+  flex: 1;
+  margin-left: 5;
+  margin-right: 5;
+  margin-top: 5;
+  margin-bottom: 5;
+`;
+
+const DisplayNameText = styled.Text`
+  fontSize: ${normalize(12)};
+  ${{ ...fonts.fontPrimary }}'
```
Comment (chinesedfan) [inline src/components/mention-area.component.js]: @binkpitch Can we simply write `${fonts.fontPrimary}` here? Please correct me if I am wrong. 😅

Comment (binkpitch) [reply src/components/mention-area.component.js]: @chinesedfan  I think ```${{ ...fonts.fontPrimary }}``` can be change to  ```${fonts.fontPrimary}```. I'll make a new commit to fix that.

File: src/components/mention-area.component.js [hunk 2]
```diff
@@ -1,36 +1,30 @@
...
+  flex-direction: row;
+  padding-top: 5;
+  padding-left: 5;
+  padding-right: 15;
+  padding-bottom: 15;
+`;
+
+const UserDetailsBox = styled.View`
+  flex: 1;
+  margin-left: 5;
```
Comment (lex111) [inline src/components/mention-area.component.js]: If just use `margin: 5;`, will it work?

Comment (binkpitch) [reply src/components/mention-area.component.js]: @lex111  When I use ```margin: 5;```, I got the following error. Seperating margin into ```margin-left```, ```margin-right```, ```margin-top```, and, ```margin-buttom``` works but I have no idea why.
screen shot 2560-10-22 at 20 35 57

Comment (chinesedfan) [reply src/components/mention-area.component.js]: @lex111 It is due to styled component's [shorthands properties restriction](https://www.styled-components.com/docs/basics#react-native).

**Comment by alejandronanez** [comment]
@binkpitch I haven't dealt with that case before, but what @chinesedfan sounds good to me.

**Comment by binkpitch** [comment]
@alejandronanez Sorry for the late reply, I just woke up. Sure, I'll make a commit of what @chinesedfan suggested.

**Comment by alejandronanez** [comment]
@binkpitch no rush at all! Let us know how this goes.

**Comment by binkpitch** [comment]
@alejandronanez I have changed Animated.View to StyledAnimatedView in my last commit. Does this look go to you?

**Pull request merged**